website/content/docs/k8s/installation/deployment-configurations/clients-outside-kubernetes.mdx

```diff
@@ -35,14 +48,68 @@ annotation `consul.hashicorp.com/auto-join-port` to an integer value or
 a named port to specify the port for the auto-join to return. This enables
 different pods to have different exposed ports.
```

Comment:
I did not need to add this annotation, even though I configured my server gossip port to be 9301. `consul members` still successfully showed all members of the cluster.

Reply:
I think it's still useful documentation to have, but that's why I didn't include this annotation in the instructions below.

Reply:
weird, I don't know how the join could have worked without this? Wouldn't it try 8301 unless explicitly configured?

Reply:
@ishustava I did change the gossip port on the server to 9301. I know its super weird I would expect it to use that code as well!

I would have expected the external client agent fail to join because it would try 8301. However, when trying the exactly the steps [here](https://github.com/hashicorp/consul-helm/pull/740) while replacing the retry-join with `-retry-join 'provider=k8s host_network=true label_selector="app=consul,component=server"'` I observed it joining just fine and on port 9301, which is definitely mind boggling. I will retry those steps one more time tomorrow morning just to make sure.

Reply:
Update: I ran through the steps with `-retry-join 'provider=k8s host_network=true label_selector="app=consul,component=server"'` on the external agent without the annotation and it works. In fact, when I do add the annotation, it doesn't accept an integer value, and if I add the port "9301" as a string, the consul clients on k8s itself fail to join saying no consul servers found. 

I think for these docs I should remove this annotation, since it doesn't seem to work. wdyt @ishustava @lkysow?

Reply:
As discussed lets:
1. Add a warning that if you're using 9301 and your consul servers are on different nodes (which is unlikely because in that case you wouldn't need to use 9301) then you can't use autojoin
1. Create a bug in go-discover about the annotation issue

Reply:
I removed the note about the annotation since its currently broken (we can add it back once the bug is fixed in go-discover), and added the note/warning about default port. I phrased it a bit differently, let me know if you think its clear or not

Reply:
Writing the bug right now